common/src/main/java/org/apache/uniffle/common/ShuffleBlockInfo.java

```diff
@@ -27,14 +29,14 @@ public class ShuffleBlockInfo {
   private int shuffleId;
   private long crc;
   private long taskAttemptId;
-  private byte[] data;
+  private ByteBuf data;
   private List<ShuffleServerInfo> shuffleServerInfos;
   private int uncompressLength;
   private long freeMemory;
 
   public ShuffleBlockInfo(int shuffleId, int partitionId, long blockId, int length, long crc,
-      byte[] data, List<ShuffleServerInfo> shuffleServerInfos,
-      int uncompressLength, int freeMemory, long taskAttemptId) {
+      ByteBuf data, List<ShuffleServerInfo> shuffleServerInfos,
+      int uncompressLength, long freeMemory, long taskAttemptId) {
     this.partitionId = partitionId;
```

Comment:
We can keep the original constructor for tests and add a new constructor for new implement.